go/vt/sqlparser/utils.go

```diff
@@ -113,3 +113,52 @@ func NormalizeAlphabetically(query string) (normalized string, err error) {
 	}
 	return String(stmt), nil
 }
+
+// ReplaceTableQualifiers takes a statement's table expressions and
+// replaces any cases of the provided database name with the
+// specified replacement name.
+// Note: both database names provided should be unescaped strings.
+func ReplaceTableQualifiers(query, olddb, newdb string) (string, error) {
+	if newdb == olddb {
+		// Nothing to do here.
+		return query, nil
+	}
+	in, err := Parse(query)
+	if err != nil {
+		return "", err
+	}
+
+	oldQualifier := NewIdentifierCS(olddb)
+	newQualifier := NewIdentifierCS(newdb)
+
+	modified := false
+	upd := Rewrite(in, func(cursor *Cursor) bool {
+		switch node := cursor.Node().(type) {
+		case TableName:
+			if !node.Qualifier.IsEmpty() &&
+				node.Qualifier.String() == oldQualifier.String() &&
+				node.Qualifier.String() != newQualifier.String() {
+				node.Qualifier = newQualifier
```

Comment:
In the start of the function, it is already checked that olddb and newdb are not same. So, `node.Qualifier.String() != newQualifier.String()` check is not needed.